authorizer-app-backend/src/main/java/org/radarbase/authorizer/doa/RestSourceUserRepositoryImpl.kt

```diff
@@ -140,6 +140,26 @@ class RestSourceUserRepositoryImpl(
         }
     }
 
+    override fun queryAllWithElapsedEndDate(
+        sourceType: String?
+    ): List<RestSourceUser> {
+        var queryString = "SELECT u FROM RestSourceUser u WHERE"
+        queryString += " u.endDate < CURRENT_TIMESTAMP"
+
+        if (sourceType != null) {
+            queryString += " AND u.sourceType = :sourceType"
+        }
+
+        return transact {
+            val query = createQuery(queryString, RestSourceUser::class.java)
+            if (sourceType != null) {
+                query.setParameter("sourceType", sourceType)
+            }
+            val users = query.resultList
+            users
```

Comment:
its fine to just return the resultList. If needed, you could add a comment that it contains users, but it looks clear already.
```suggestion
            query.resultList
```